### PR TITLE
[#3203] improvement(PyClient): Add pylint Rules for Python client

### PR DESCRIPTION
**Pull request by noidname01**
### What changes were proposed in this pull request?

* Add pylint rule for logging, logging should use old format (%), not fstring
* Unify all the logging format
* Set many disable rules for future modification (I add TODO tags after them, will work on them after this PR)

### Why are the changes needed?

Fix: #3203 

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?
```bash
./gradlew :clients:client-python:test
```

### CODE REVIEW

**Comment by noidname01** [comment]
@xunliu Please help me to review this.

**Comment by jerryshao** [comment]
Hi @noidname01 thanks a lot for your contribution. I would suggest to ingrate python lint command with Gradle, so that we can Gradle command to run them all, no matter it is java, python or others.

Besides, we use spotless for JVM language lint, I also found that it can also support python (https://github.com/diffplug/spotless?tab=readme-ov-file), can we use spotless? WDYT @xunliu

**Comment by noidname01** [comment]
Hi @jerryshao, I have integrated pylint into Gradle.
For the use of spotless, the most of big python project communities (Ray, Tensorflow, ...) still use pylint or flake8 as linter, so I think it's better to follow this best practice.

**Comment by jerryshao** [comment]
> Hi @jerryshao, I have integrated pylint into Gradle. For the use of spotless, the most of big python project communities (Ray, Tensorflow, ...) still use pylint or flake8 as linter, so I think it's better to follow this best practice.

OK, I see, thanks for the explanation.

File: clients/client-python/gravitino/dto/dto_converters.py
Comment (xunliu) [inline clients/client-python/gravitino/dto/dto_converters.py:30]: Why need change `elif` to `if`?

File: clients/client-python/pylintrc
Comment (xunliu) [inline clients/client-python/pylintrc:38]: Will these TODO list fix in the one by one?

File: clients/client-python/tests/integration/integration_test_env.py
Comment (xunliu) [inline clients/client-python/tests/integration/integration_test_env.py:49]: Is `#` the recommend comment format for Python?

File: clients/client-python/README.md
Comment (noidname01) [reply clients/client-python/README.md]: Got it, I forgot to delete this.

File: clients/client-python/tests/integration/integration_test_env.py
Comment (noidname01) [reply clients/client-python/tests/integration/integration_test_env.py:49]: It seems that pylint think this is a long string, not a comment. Maybe try to add new line in this?

File: clients/client-python/pylintrc
Comment (noidname01) [reply clients/client-python/pylintrc:38]: Yes, I plan to do so.

File: clients/client-python/gravitino/dto/dto_converters.py
Comment (noidname01) [reply clients/client-python/gravitino/dto/dto_converters.py:30]: This is suggested by pylint, the no-else-return rule
https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/no-else-return.html

File: clients/client-python/tests/integration/integration_test_env.py
Comment (noidname01) [reply clients/client-python/tests/integration/integration_test_env.py:49]: https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/pointless-string-statement.html

Maybe we need to determine whether to disable this rule to allow long string comment.

File: clients/client-python/gravitino/utils/http_client.py
Comment (noidname01) [reply clients/client-python/gravitino/utils/http_client.py:121]: This change is because I found there's a usage `_request("close")` and pass no endpoint as argument, and it make pylint alert.
Maybe modify the usage into `_request("close", None)`?

Comment (zhaoyongjie) [reply clients/client-python/gravitino/utils/http_client.py:121]: I'm not sure how "close"  works here...but the endpoint argument should be passed a value for constructing a URL, so IMO, a `None` value is a invalid value.

Comment (noidname01) [reply clients/client-python/gravitino/utils/http_client.py:121]: Here is the close function:

https://github.com/datastrato/gravitino/blob/b841e58ad3520f4750aa2c6fe52ae97d98a7ccbd/clients/client-python/gravitino/utils/http_client.py#L161-L162

File: clients/client-python/build.gradle.kts
Comment (jerryshao) [inline clients/client-python/build.gradle.kts:44]: How do we trigger the pylint task for gradle, is it `./gradlew pylint`?

File: clients/client-python/gravitino/utils/http_client.py
Comment (noidname01) [reply clients/client-python/gravitino/utils/http_client.py]: I've replaced `self._request("close")` with `self._request("close", "/")` , meaning the endpoint is the host url itself.

File: clients/client-python/tests/integration/integration_test_env.py
Comment (noidname01) [reply clients/client-python/tests/integration/integration_test_env.py:49]: I think the comment should use `#` and the docstring use `"""`. The case here is more like a comment.

File: clients/client-python/gravitino/utils/http_client.py
Comment (xunliu) [inline clients/client-python/gravitino/utils/http_client.py:69]: Why we need deleted these codes?

Comment (noidname01) [reply clients/client-python/gravitino/utils/http_client.py:69]: Pylint thinks it is redundant with no-else-return rule. The default return of Python function is `None`, there is no need to return explicitly.

Comment (xunliu) [reply clients/client-python/gravitino/utils/http_client.py:69]: OK, Thank you for your reply.

**Review by xunliu — APPROVED** [review]
@noidname01 Thank you for your contributions.
LGTM

File: clients/client-python/build.gradle.kts
Comment (jerryshao) [reply clients/client-python/build.gradle.kts:44]: I think this should not be resolved unless we have a consensus on this. 

I would suggest that we can trigger the pylint task implicitly before compile or other task, like spotless which will be triggered automatically when running `compile`, `build` or others.

Here you bring in a new task/command without documentation, it is not a good practice.

**Comment by jerryshao** [comment]
@xunliu I think there are left questions should be answered before merging, no need to hurry. Besides, when you merge a PR, please **follow the process** in `MAINTAINERS.md`.

**Comment by jerryshao** [comment]
@noidname01 normally, if the comment is a question that should be answered, it would be better not to resolve this comment yourself until we get a consensus on it, otherwise, some unaddressed problems will be ignored by other reviewers.

**Comment by noidname01** [comment]
@jerryshao Got it, sorry for the inconveniences

File: clients/client-python/build.gradle.kts
Comment (noidname01) [reply clients/client-python/build.gradle.kts:44]: Actually it will run before running integration tests of python. The `test` task depends on the `pylint` task. I will add pylint task before build task if necessary.

Comment (jerryshao) [reply clients/client-python/build.gradle.kts:44]: @noidname01 sorry for late response because of public holiday. I would suggest we align with spotless, can be triggered before `compile`?

**Comment by jerryshao** [comment]
@noidname01 would you please submit a new cherry-pick PR to backport this to `branch-0.5`?

**Comment by noidname01** [comment]
@jerryshao Done.
- [x] #3279

**Pull request merged**
